### ISSUE TITLE
RuntimeError: neuronx-cc failed with 1 when try to compile Stable Diffusion v2

### ISSUE THREAD

**Issue by gokerguner** [aws-neuron/aws-neuron-sdk#690]
I'm trying to compile Stable Diffusion v2 with this code: [sd2_512_compile.py](https://github.com/aws-neuron/aws-neuron-sdk/blob/master/src/benchmark/pytorch/sd2_512_compile.py)

and run this code: `python3 sd2_512_compile.py`
AMI Name: Deep Learning AMI Neuron PyTorch 1.13 (Ubuntu 20.04) in predefined conda env: aws_neuron_venv_pytorch

neuronx-cc                    2.6.0.19+3d819e565
torch-neuronx               1.13.1.1.7.0

I don't know what else packages that I should provide its version. I can give more knowledge if you need. And the whole error message is:

```2023-06-15T12:11:09Z ERROR 85724 [Codegen]: Instruction LoadActFuncSet I-50114-4-PWP generates invalid ISA instruction: 
ActivationTableLoad
inst failed assertion check: 'valid_act_tbl_sel'
2023-06-15T12:11:09Z ERROR 85724 [WalrusDriver]: An exception was thrown:
--------------------------------------------------------------------------------
 0# __cxa_throw in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/jobs/support/../../../starfish/lib/libwalrus.so
 1# void neuronxcc::walrus::SundaGen::checkISAValidity<SUNDA_ISA_TPB_CTRL_NO_STRUCT>(SUNDA_ISA_TPB_CTRL_NO_STRUCT const&, bir::Instruction const&) in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/jobs/support/../../../starfish/lib/libwalrus.so
 2# neuronxcc::walrus::SundaGen::visitInstLoadActFuncSet(bir::InstLoadActFuncSet&) in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/jobs/support/../../../starfish/lib/libwalrus.so
 3# neuronxcc::walrus::SundaGen::SundaGen(neuronxcc::walrus::Logger&, bir::Module&, neuronxcc::walrus::Generator::CodeGenMode, unsigned int) in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/jobs/support/../../../starfish/lib/libwalrus.so
 4# neuronxcc::walrus::Codegen::run(bir::Module&) in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/jobs/support/../../../starfish/lib/libwalrus.so
 5# neuronxcc::walrus::WalrusPass::run(std::vector<std::unique_ptr<bir::Module, std::default_delete<bir::Module> >, std::allocator<std::unique_ptr<bir::Module, std::default_delete<bir::Module> > > >&) in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/jobs/support/../../../starfish/lib/libwalrus.so
 6# 0x00007FE67C193E5A in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/jobs/support/../../../starfish/lib/libwalrus.so
 7# run_walrus_driver(int, char**) in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/jobs/support/../../../starfish/lib/libwalrus.so
 8# 0x00007FE6EEFAF130 in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/jobs/support/EmbeddedWalrusDriver.cpython-38-x86_64-linux-gnu.so
 9# 0x00007FE690E8EA60 in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/jobs/WalrusDriver.cpython-38-x86_64-linux-gnu.so
10# 0x00007FE690E960AE in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/jobs/WalrusDriver.cpython-38-x86_64-linux-gnu.so
11# _PyObject_MakeTpCall in /opt/aws_neuron_venv_pytorch/bin/python3.8
12# _PyObject_FastCallDict in /opt/aws_neuron_venv_pytorch/bin/python3.8
13# _PyObject_Call_Prepend in /opt/aws_neuron_venv_pytorch/bin/python3.8
14# 0x00007FE690E8CC2C in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/jobs/WalrusDriver.cpython-38-x86_64-linux-gnu.so
15# 0x00007FE690EBDD52 in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/jobs/WalrusDriver.cpython-38-x86_64-linux-gnu.so
16# _PyObject_MakeTpCall in /opt/aws_neuron_venv_pytorch/bin/python3.8
17# _PyObject_FastCallDict in /opt/aws_neuron_venv_pytorch/bin/python3.8
18# _PyObject_Call_Prepend in /opt/aws_neuron_venv_pytorch/bin/python3.8
19# 0x00007FE726068C3C in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/Job.cpython-38-x86_64-linux-gnu.so
20# 0x00007FE72607D2D6 in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/Job.cpython-38-x86_64-linux-gnu.so
21# _PyObject_MakeTpCall in /opt/aws_neuron_venv_pytorch/bin/python3.8
22# _PyObject_FastCallDict in /opt/aws_neuron_venv_pytorch/bin/python3.8
23# _PyObject_Call_Prepend in /opt/aws_neuron_venv_pytorch/bin/python3.8
24# 0x00007FE726068C3C in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/Job.cpython-38-x86_64-linux-gnu.so
25# 0x00007FE726078AC8 in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/Job.cpython-38-x86_64-linux-gnu.so
26# 0x00007FE690E97172 in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/jobs/WalrusDriver.cpython-38-x86_64-linux-gnu.so
27# _PyObject_MakeTpCall in /opt/aws_neuron_venv_pytorch/bin/python3.8
28# _PyObject_FastCallDict in /opt/aws_neuron_venv_pytorch/bin/python3.8
29# _PyObject_Call_Prepend in /opt/aws_neuron_venv_pytorch/bin/python3.8
30# 0x00007FE726095C6B in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/Pipeline.cpython-38-x86_64-linux-gnu.so
31# 0x00007FE726098082 in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/Pipeline.cpython-38-x86_64-linux-gnu.so
32# _PyObject_MakeTpCall in /opt/aws_neuron_venv_pytorch/bin/python3.8
33# _PyObject_FastCallDict in /opt/aws_neuron_venv_pytorch/bin/python3.8
34# _PyObject_Call_Prepend in /opt/aws_neuron_venv_pytorch/bin/python3.8
35# 0x00007FE726068C3C in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/Job.cpython-38-x86_64-linux-gnu.so
36# 0x00007FE72607D2D6 in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/Job.cpython-38-x86_64-linux-gnu.so
37# _PyObject_MakeTpCall in /opt/aws_neuron_venv_pytorch/bin/python3.8
38# _PyObject_FastCallDict in /opt/aws_neuron_venv_pytorch/bin/python3.8
39# _PyObject_Call_Prepend in /opt/aws_neuron_venv_pytorch/bin/python3.8
40# 0x00007FE726068C3C in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/Job.cpython-38-x86_64-linux-gnu.so
41# 0x00007FE726078AC8 in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/Job.cpython-38-x86_64-linux-gnu.so
42# _PyObject_MakeTpCall in /opt/aws_neuron_venv_pytorch/bin/python3.8
43# _PyObject_FastCallDict in /opt/aws_neuron_venv_pytorch/bin/python3.8
44# _PyObject_Call_Prepend in /opt/aws_neuron_venv_pytorch/bin/python3.8
45# 0x00007FE725B2030C in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/commands/CompileCommand.cpython-38-x86_64-linux-gnu.so
46# 0x00007FE725B59D0E in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/commands/CompileCommand.cpython-38-x86_64-linux-gnu.so
47# _PyObject_MakeTpCall in /opt/aws_neuron_venv_pytorch/bin/python3.8
48# _PyObject_FastCallDict in /opt/aws_neuron_venv_pytorch/bin/python3.8
49# _PyObject_Call_Prepend in /opt/aws_neuron_venv_pytorch/bin/python3.8
50# 0x00007FE725B26971 in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/commands/CompileCommand.cpython-38-x86_64-linux-gnu.so
51# _PyObject_MakeTpCall in /opt/aws_neuron_venv_pytorch/bin/python3.8
52# _PyObject_FastCallDict in /opt/aws_neuron_venv_pytorch/bin/python3.8
53# _PyObject_Call_Prepend in /opt/aws_neuron_venv_pytorch/bin/python3.8
54# 0x00007FE726169C4C in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/CommandDriver.cpython-38-x86_64-linux-gnu.so
55# 0x00007FE72617BB9A in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/CommandDriver.cpython-38-x86_64-linux-gnu.so
56# _PyObject_MakeTpCall in /opt/aws_neuron_venv_pytorch/bin/python3.8
57# _PyObject_FastCallDict in /opt/aws_neuron_venv_pytorch/bin/python3.8
58# _PyObject_Call_Prepend in /opt/aws_neuron_venv_pytorch/bin/python3.8
59# 0x00007FE72616C14D in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/CommandDriver.cpython-38-x86_64-linux-gnu.so
60# 0x00007FE72616C322 in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/CommandDriver.cpython-38-x86_64-linux-gnu.so
61# 0x00007FE726174E55 in /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/neuronxcc/driver/CommandDriver.cpython-38-x86_64-linux-gnu.so
62# _PyObject_MakeTpCall in /opt/aws_neuron_venv_pytorch/bin/python3.8
63# _PyEval_EvalFrameDefault in /opt/aws_neuron_venv_pytorch/bin/python3.8
64# _PyEval_EvalCodeWithName in /opt/aws_neuron_venv_pytorch/bin/python3.8
65# PyEval_EvalCode in /opt/aws_neuron_venv_pytorch/bin/python3.8
66# 0x000000000067DBF1 in /opt/aws_neuron_venv_pytorch/bin/python3.8
67# 0x000000000067DC6F in /opt/aws_neuron_venv_pytorch/bin/python3.8
68# 0x000000000067DD11 in /opt/aws_neuron_venv_pytorch/bin/python3.8
69# PyRun_SimpleFileExFlags in /opt/aws_neuron_venv_pytorch/bin/python3.8
70# Py_RunMain in /opt/aws_neuron_venv_pytorch/bin/python3.8
71# Py_BytesMain in /opt/aws_neuron_venv_pytorch/bin/python3.8
72# __libc_start_main in /lib/x86_64-linux-gnu/libc.so.6
73# _start in /opt/aws_neuron_venv_pytorch/bin/python3.8
--------------------------------------------------------------------------------
2023-06-15T12:11:09Z ERROR 85724 [WalrusDriver]: Walrus pass: codegen failed!
2023-06-15T12:11:09Z ERROR 85724 [WalrusDriver]: Failure Reason: ISA check failed
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]: ***************************************************************
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:  An Internal Compiler Error has occurred
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]: ***************************************************************
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]: 
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]: Error message:  Walrus driver failed to complete
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]: 
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]: Error class:    AssertionError
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]: Error location: Unknown
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]: Command line:   /opt/aws_neuron_venv_pytorch/bin/neuronx-cc compile sd2_compile_dir_512/unet/model --framework XLA --target trn1 --output sd2_compile_dir_512/unet/graph.neff --model-type=unet-inference
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]: 
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]: Internal details:
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   File "neuronxcc/driver/CommandDriver.py", line 259, in neuronxcc.driver.CommandDriver.CommandDriver.run
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   File "neuronxcc/driver/commands/CompileCommand.py", line 1089, in neuronxcc.driver.commands.CompileCommand.CompileCommand.run
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   File "neuronxcc/driver/commands/CompileCommand.py", line 1040, in neuronxcc.driver.commands.CompileCommand.CompileCommand.runPipeline
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   File "neuronxcc/driver/commands/CompileCommand.py", line 1065, in neuronxcc.driver.commands.CompileCommand.CompileCommand.runPipeline
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   File "neuronxcc/driver/commands/CompileCommand.py", line 1069, in neuronxcc.driver.commands.CompileCommand.CompileCommand.runPipeline
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   File "neuronxcc/driver/Job.py", line 300, in neuronxcc.driver.Job.SingleInputJob.run
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   File "neuronxcc/driver/Job.py", line 326, in neuronxcc.driver.Job.SingleInputJob.runOnState
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   File "neuronxcc/driver/Pipeline.py", line 30, in neuronxcc.driver.Pipeline.Pipeline.runSingleInput
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   File "neuronxcc/driver/jobs/WalrusDriver.py", line 232, in neuronxcc.driver.jobs.WalrusDriver.WalrusDriver.run
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   File "neuronxcc/driver/Job.py", line 300, in neuronxcc.driver.Job.SingleInputJob.run
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   File "neuronxcc/driver/Job.py", line 326, in neuronxcc.driver.Job.SingleInputJob.runOnState
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   File "neuronxcc/driver/jobs/WalrusDriver.py", line 706, in neuronxcc.driver.jobs.WalrusDriver.WalrusDriver.runSingleInput
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]: 
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]: Version information:
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   NeuronX Compiler version 2.6.0.19+3d819e565
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   Python version 3.8.10
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   HWM version 2.6.0.0-826e77395
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   NEFF version Dynamic
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   TVM not available
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   NumPy version 1.21.6
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]:   MXNet not available
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]: 
2023-06-15T12:11:12Z ERROR 85724 [neuronx-cc]: Artifacts stored in: /home/ubuntu/neuronxcc-5u7s427v
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /home/ubuntu/sd2_512_compile.py:179 in <module>                                                  │
│                                                                                                  │
│   176 encoder_hidden_states_1b = torch.randn([1, 77, 1024]).bfloat16()                           │
│   177 example_inputs = sample_1b, timestep_1b, encoder_hidden_states_1b                          │
│   178                                                                                            │
│ ❱ 179 unet_neuron = torch_neuronx.trace(                                                         │
│   180 │   unet,                                                                                  │
│   181 │   example_inputs,                                                                        │
│   182 │   compiler_workdir=os.path.join(COMPILER_WORKDIR_ROOT, 'unet'),                          │
│                                                                                                  │
│ /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/torch_neuronx/xla_impl/trace.py:309 in  │
│ trace                                                                                            │
│                                                                                                  │
│   306 │   │   │   handle.write(hlo.SerializeToString())                                          │
│   307 │   │                                                                                      │
│   308 │   │   # Compile HLO to NEFF                                                              │
│ ❱ 309 │   │   neff_filename = hlo_compile(model_dir, compiler_workdir, compiler_args)            │
│   310 │   │                                                                                      │
│   311 │   │   # Read the NEFF from disk                                                          │
│   312 │   │   with open(neff_filename, 'rb') as handle:                                          │
│                                                                                                  │
│ /opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/torch_neuronx/xla_impl/trace.py:232 in  │
│ hlo_compile                                                                                      │
│                                                                                                  │
│   229 │                                                                                          │
│   230 │   status = subprocess.run(command).returncode                                            │
│   231 │   if status != 0:                                                                        │
│ ❱ 232 │   │   raise RuntimeError(f'neuronx-cc failed with {status}')                             │
│   233 │                                                                                          │
│   234 │   return neff_filename                                                                   │
│   235                                                                                            │
╰──────────────────────────────────────────────────────────────────────────────────────���───────────╯
RuntimeError: neuronx-cc failed with 1```

**Comment by aws-rhsoln** [comment]
Thank you for creating an issue. We are looking into this and would provide an update once we have a root cause.

**Comment by awsilya** [comment]
@gokerguner sorry it took a while.  We believe the issue has been fixed in 2.12 release.  Could you please try it and let us know.

Thank you.
